test/test_joint_trajectory_splitter.py

```diff
@@ -0,0 +1,305 @@
+#! /usr/bin/env python
+
+import rospy
+import actionlib
+import control_msgs.msg
+import roslaunch
+import rospkg
+import trajectory_msgs.msg
+import pytest
+import copy
+from giskardpy import logging
+
+@pytest.fixture(scope=u'function')
+def init_ros():
+    rospy.init_node('JointTrajectorySplitterTest')
+
+
+@pytest.fixture(scope=u'function')
+def giskard_package_path():
+    rospack = rospkg.RosPack()
+    giskard_package_path = rospack.get_path('giskardpy')
+    return giskard_package_path
+
+@pytest.fixture(scope=u'function')
+def simple_trajectory_goal():
+    goal = control_msgs.msg.FollowJointTrajectoryGoal()
+    goal.trajectory.joint_names = ["joint1", "joint2", "joint3", "joint4"]
+
+    point1 = trajectory_msgs.msg.JointTrajectoryPoint()
+    point1.positions.extend([1.0, 1.0, 1, 0])
+    point1.time_from_start = rospy.Time(0)
+    goal.trajectory.points.append(point1)
+
+    point2 = copy.copy(point1)
+    point2.time_from_start = rospy.Time(1)
+    goal.trajectory.points.append(point2)
+
+    point3 = copy.copy(point1)
+    point3.time_from_start = rospy.Time(2)
+    goal.trajectory.points.append(point3)
+
+    return goal
+
+@pytest.fixture(scope=u'module')
+def missing_joint_goal():
+    goal = control_msgs.msg.FollowJointTrajectoryGoal()
+    goal.trajectory.joint_names = ["joint1", "joint3", "joint4"]
+
+    point1 = trajectory_msgs.msg.JointTrajectoryPoint()
+    point1.positions.extend([1.0, 1.0, 1, 0])
+    point1.time_from_start = rospy.Time(0)
+    goal.trajectory.points.append(point1)
+
+    point2 = copy.copy(point1)
+    point2.time_from_start = rospy.Time(1)
+    goal.trajectory.points.append(point2)
+
+    point3 = copy.copy(point1)
+    point3.time_from_start = rospy.Time(2)
+    goal.trajectory.points.append(point3)
+
+    return goal
+
+@pytest.fixture(scope=u'module')
+def long_trajectory_goal():
+    goal = control_msgs.msg.FollowJointTrajectoryGoal()
+    goal.trajectory.joint_names = ["joint1",'joint2', "joint3", "joint4"]
+
+    point1 = trajectory_msgs.msg.JointTrajectoryPoint()
+    point1.positions.extend([1.0, 1.0, 1, 0])
+    point1.time_from_start = rospy.Time(0)
+    goal.trajectory.points.append(point1)
+
+    point2 = copy.copy(point1)
+    point2.time_from_start = rospy.Time(10)
+    goal.trajectory.points.append(point2)
+
+    point3 = copy.copy(point1)
+    point3.time_from_start = rospy.Time(20)
+    goal.trajectory.points.append(point3)
+
+    return goal
+
+
+@pytest.fixture(scope=u'function')
+def launch_timeout_test_nodes(request):
+    launch = roslaunch.scriptapi.ROSLaunch()
+    launch.start()
```

Comment:
There is quite a bit of redundancy in these functions. Maybe you can somehow use these fixture dependencies to save some code as well?